cmd/provisioner-localpv/app/provisioner_blockdevice.go

```diff
@@ -84,17 +86,26 @@ func (p *Provisioner) ProvisionBlockDevice(opts pvController.VolumeOptions, volu
 	//labels[string(v1alpha1.StorageClassKey)] = *className
 
 	//TODO Change the following to a builder pattern
-	pvObj, err := mPV.NewBuilder().
+	pvObjBuilder := mPV.NewBuilder().
 		WithName(name).
 		WithLabels(labels).
 		WithAnnotations(volAnnotations).
 		WithReclaimPolicy(opts.PersistentVolumeReclaimPolicy).
 		WithAccessModes(pvc.Spec.AccessModes).
-		WithVolumeMode(fs).
+		WithVolumeMode(volumeMode).
 		WithCapacityQty(pvc.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]).
-		WithLocalHostPathFormat(path, fsType).
-		WithNodeAffinity(nodeHostname).
-		Build()
+		//WithLocalHostPathFormat(path, fsType).
+		WithNodeAffinity(nodeHostname)
+
+	// If volumeMode set to "Block", then provide the blockDevicePath
+	if *opts.PVC.Spec.VolumeMode == v1.PersistentVolumeBlock {
+		pvObjBuilder.WithLocalHostPathFormat(blkPath, "")
```

Comment:
Set VolumeMode also here on the builder object. Or add a new builder function called - WithLocalHostBlockDevice and set the mode within. Similarly update the builder function to set the pv fstype.